tests/integration/framework/framework.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	tu "istio.io/istio/tests/util"
+)
+
+// runnable is used for Testing purposes.
+type runnable interface {
+	Run() int
+}
+
+// IstioTestFramework is core test framework struct
+type IstioTestFramework struct {
+	TestEnv    TestEnv
+	TestID     string
+	Components []Component
+}
+
+// NewIstioTestFramework create a IstioTestFramework with a given environment and ID
+func NewIstioTestFramework(env TestEnv, id string) *IstioTestFramework {
+	return &IstioTestFramework{
+		TestEnv:    env,
+		Components: env.GetComponents(),
+		TestID:     id,
+	}
+}
+
+// SetUp sets up the whole environment as well brings up components
+func (framework *IstioTestFramework) SetUp() (err error) {
+	if err = framework.TestEnv.Bringup(); err != nil {
+		log.Printf("Failed to bring up environment")
+		return
+	}
+	for _, comp := range framework.Components {
+		if err := comp.Start(); err != nil {
+			log.Printf("Failed to setup component: %s", comp.GetName())
+			return err
+		}
+	}
+
+	if ready, err := framework.IsEnvReady(); err != nil || !ready {
+		err = fmt.Errorf("failed to get env ready: %s", err)
+		return err
+	}
+
+	// TODO: Find more reliable way to tell if local components are ready to serve
+	time.Sleep(3 * time.Second)
```

Comment:
I think we should define that IsAlive() returns true when it is ready to serve.

So you can either push the wait inside the Component or have the Component implement proper health check.